bitcoin/src/blockdata/transaction.rs

```diff
@@ -1173,6 +1171,30 @@ impl From<&Transaction> for Wtxid {
     fn from(tx: &Transaction) -> Wtxid { tx.wtxid() }
 }
 
+/// Returns the effective_value `Amount` returning `None` on overflow.
+///
+/// The effective value is the value of this output minus the amount to spend it.  That is, the
+/// effective_value can be calculated as value - (fee_rate * weight).
+///
+/// Note: the effective value of a [`Transaction`] may increase more than the effective value of
+/// a [`TxOut`] when adding another [`TxOut`] to the transaction.  This happens when the new
+/// [`TxOut`] added causes the output length `VarInt` to increase its encoding length.
+///
+/// # Arguments
+///
+/// * `fee_rate` - `FeeRate`in sat/kwu
+/// * `satisfaction_weight` - satisfied spending conditions weight
+// Note, this method could return Result instead of Option for richer errors.
+pub fn effective_value(
+    fee_rate: FeeRate,
+    satisfaction_weight: Weight,
+    value: Amount,
+) -> Option<SignedAmount> {
+    let weight = satisfaction_weight.checked_add(TxIn::BASE_WEIGHT)?;
+    let signed_input_fee = fee_rate.checked_mul_by_weight(weight)?.to_signed().ok()?;
+    value.to_signed().ok()?.checked_sub(signed_input_fee)
```

Comment:
In theory, this doesn't allow very large amounts but it's OK since there are only 21000000 bitcoins. :)

Reply:
Yeah singed has one less significant digit to use to store numbers since it is instead used to represent negative or positive which is I think what you're pointing out here.  I can't really think of when this would be a problem since as you say the amount is limited at 21 mil.